core/server/master/src/main/java/alluxio/master/file/meta/InodeLockManager.java

```diff
@@ -46,9 +46,9 @@
    */
   public final LoadingCache<Long, WeakSafeReentrantReadWriteLock> mInodeLocks =
       CacheBuilder.<Long, WeakSafeReentrantReadWriteLock>newBuilder()
-          .weakValues()
+          .maximumSize(1000000)
```

Comment:
@aaudiber 

Is this required to have weakValues for correctness? Is it required to be unlimited size?

Reply:
We don't necessarily require weak values or unlimited size, but we do require that locks only be removed from the cache when nothing holds a reference to them. Otherwise two threads can successfully write lock the same inode because the lock was evicted after the first thread grabbed it.

For eviction to be a valid strategy, we need to do reference counting ourselves and only evict unreferenced locks.